website/docs/reference/deploy/environment-import-export.mdx

```diff
@@ -0,0 +1,110 @@
+
+When exporting, you select a set of features and **one** environment to export the configuration from. The environment must be the same for all features.
+
+Then, when you import, you must select **one** environment and **one** project to import into. All features are imported into that project in that environment. If Unleash is unable to import the configuration safely, it will tell you why the import failed and what you need to do fix it (read more about [import rejection](#import-rejection).
+
+## Import & Export items
+
+When you export features, the export will contain both feature-specific configuration and global configuration.
+
+On the project-level these  items are exported:
+
+* [the feature itself](../feature-toggles.mdx)
+* [feature tags](../tags.md)
+
+On the environment-level, these items are exported for the chosen environment:
+
+* [activation strategies](../activation-strategies.md) including [constraints](../strategy-constraints.md) and references to [segments](../segments.mdx)
+* [variants](../feature-toggle-variants.md)
+* enabled/disabled
+
+Additionally, these global configuration items are exported:
+* [custom context fields](../unleash-context.md#context-fields)
+* [feature tag types](../tags.md#tag-types)
+
+Importantly, while references to [segments](../segments.mdx)) are exported, the segments themselves are **not** exported. Consult the [import rejection](#import-rejection) section for more information.
+
+## Export
+
+You can export features either from a project search or the global feature search. Use the search functionality to narrow the results to the list of features you want to export and use the export button to select environment. All features included in your search results will be included in the export.
+
+<Figure caption="Search-driven export screen with WYSIWYG mode."  img="/img/export.png"/>
+
+## Import
+
+Import is a 3 stage process designed to be efficient and error-resistant.
+
+### Import stages
+
+* **upload** - you can upload previously exported JSON file or copy-paste export data from the exported JSON file into the code editor
+* **validation** - you will get feedback on any errors or warnings before you do the actual import. This makes sure your feature flags configurations. You will not be able to finish the import if you have errors. Warnings don't stop you from importing.
+* **import** - the actual import that creates a new configuration in the target environment or creates a [change request](../change-requests.md) when the environment has change requests enabled
+
+![Three-staged import process](/img/import.png)
+
+### Import rejection
+
+An import will be rejected when there's a possibility of data corruption.
+
+#### Context fields
+
+The exported and imported context fields have the same name but the target environment doesn't support some of the source environment context field values. If the context fields don't exist in the target environment they will be created.
```

Comment:
I still don't understand this. What are the imported context fields and what are the exported context fields? Is this about conflicts between context fields that you are trying to import and context fields that already exist in the Unleash instance that you're trying to import them into?

I think there's three cases here, right? (Note: the "new" context field is the imported one, the "old" context field is the one that already exists in the instance)
1. You import a context field that doesn't exist in the target instance. All good.
2. You import a context field that already exists in the target instance. The new and the old context fields either have the exact same legal values or both accept everything. This is fine.
3. You import a context field that already exists in the target instance. There is a mismatch between the old and the new one. One of:
  i. The new context field has no restrictions on the value, but the existing one has a defined set of legal values
  ii. The old context field has no restrictions on the value, but the new field has a defined set of legal values
  iii. Both fields have legal values, but they are **not exactly the same** (even if the new custom context field's values are a strict subset of the old field's values)

Is that correct?

Reply:
The actual algorithms is much simpler: https://github.com/Unleash/unleash/blob/main/src/lib/features/export-import-toggles/import-context-validation.ts
Here's the test for it: https://github.com/Unleash/unleash/blob/main/src/lib/features/export-import-toggles/import-context-validation.test.ts

In essence, if you're importing legal values that don't exist in the target instance you get an error.

Reply:
Yeah, okay, so it's more like two:
1. The imported field doesn't exist already. All good.
2. If the imported context field has legal values, then the existing context field must also have at least those values defined. Any other values in the existing field is also okay.

Reply:
```suggestion
When you import a custom context field **with legal values defined**:

If a custom context field with the same name already exists in the target instance, then the pre-existing context field must have **at least** those legal values defined. In other words, the imported context field legal values must be a subset of the already existing context field's legal values.

When you import custom context fields without legal values or custom context fields that don't already exist in the target instance, then there are no requirements. 

Custom context fields that don't already exist in the target instance will be created upon import.
```